src/Blazored.Modal/ModalReference.cs

```diff
@@ -11,11 +11,24 @@ public class ModalReference
 
         private Action<ModalResult> Closed;
 
-        public ModalReference(Guid modalInstanceId, RenderFragment modalInstance)
+        private IModalService ModalService { get; }
```

Comment:
This can be made a `readonly` field, I don't think this needs to be a property. Also we can just type it as `ModalService` (see other comment on `IModalService` changes).
```suggestion
        private readonly ModalService _modalService { get; }
```

Reply:
While we are at it, should we also make the Closed action and the _resultCompletion readonly? They are both private and not written to anywhere but the constructor. 

Maybe this should be in a seperate PR?

Reply:
I agree, I only didn't raise it as it wasn't directly to do with your work. But as you're making changes there any way it would make sense to do it at the same time.